assets/src/edit-story/components/panels/textStyle/color.js

```diff
@@ -74,6 +89,31 @@ function ColorControls({ selectedElements, onSetProperties }) {
           label={__('Background color', 'web-stories')}
         />
       </Row>
+      {backgroundColor && (
+        <Row>
+          <Space flex="1" />
+          <ToggleButton
+            icon={<FillIcon width={32} height={32} />}
+            value={state.backgroundType === 'fill'}
+            isMultiple={false}
+            label={__('Fill', 'web-stories')}
+            onChange={(value) =>
+              setState({ ...state, backgroundType: value ? 'fill' : '' })
```

Comment:
Just noticed [this demo](https://xwp.slack.com/archives/CCCAUAH6F/p1585060989330600?thread_ts=1585031022.322800&cid=CCCAUAH6F). We **will** need three states for background fill type, with "none" being default. Please add this extra button for transparent fill and maybe create some constants for the 3 values.

Also, we then need a data migration setting background fill type for all previous text elements based on whether or not they have a background color. If background is currently null or transparent color, set to `NONE`, otherwise set to `FILL`.

Reply:
And of course use these new values in output component too.

Reply:
Yeah, sounds really reasonable add a NONE state. About data migration, what do you mean is to add a hardcoded condition based on the rules suggested above by you?

Reply:
Yes, add a new migration to `migrations` in the root folder and make sure to set proper background fill type on all old elements in old stories.